File: 백준/Silver/4134. 다음 소수/다음 소수.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int t = Integer.parseInt(br.readLine());
        StringBuilder sb = new StringBuilder();

        for (int i = 0; i < t; i++) {
            long num = Long.parseLong(br.readLine());

            while (true) {
                if (isPrime(num)) {
                    break;
                }
                num++;
            }
            sb.append(num).append("\n");
        }

        bw.write(sb.toString());
        bw.flush();
        bw.close();
    }

    private static boolean isPrime(long num) {
        if (num < 2) {
            return false;
        } else if (num == 2) {
            return true;
        } else if (num % 2 == 0) {
            return false;
        } else {
            for (long i = 3; i <= Math.sqrt(num); i += 2) {
                if(num % i == 0) {
                    return false;
                }
            }
            return true;
        }
    }
}

File: 백준/Silver/4134. 다음 소수/다음 소수.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import kotlin.math.sqrt

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    val t = br.readLine().toInt()
    val sb = StringBuilder()
    repeat(t) {
        var num = br.readLine().toLong()

        while (true) {
            if (isPrime(num)) {
                break
            }
            num++
        }
        sb.append(num).append("\n")
    }

    bw.write(sb.toString())
    bw.flush()
    bw.close()
}

fun isPrime(num: Long): Boolean {
    if (num < 2) {
        return false
    }else if (num == 2L) {
        return true
    }else if (num % 2 == 0L) {
        return false
    } else {
        for(i in 3..sqrt(num.toDouble()).toLong() step 2) {
            if (num % i == 0L) {
                return false
            }
        }
        return true
    }
}